Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -435,10 +436,11 @@ WarpXOpenPMDPlot::DumpToFile (WarpXParticleContainer* pc,
            auto const positionComponents = detail::getParticlePositionComponentLabels();
 #if defined(WARPX_DIM_RZ)
            {
-              std::shared_ptr<amrex::ParticleReal> z(
-                      new amrex::ParticleReal[numParticleOnTile],
-                      [](amrex::ParticleReal const *p) { delete[] p; }
-              );
+               //for std::make_shared<T[]> we should wait C++20
+               //in the meanwhile, a unique_ptr can be converted in a shared_ptr
+               std::shared_ptr<amrex::ParticleReal[]> z =
+                std::make_unique<amrex::ParticleReal[]>(numParticleOnTile);
+
```

Comment:
This solution leaves me a bit unhappy. But we can't make it cleaner until C++20. If it is too misleading we might go back to the original version.